src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,203 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private static class Node<T> {
+        T element;
+        Node<T> next;
+        Node<T> previous;
+
+        public Node(Node<T> previous, T element, Node<T> next) {
+            this.previous = previous;
+            this.element = element;
+            this.next = next;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newNode = new Node<>(null, value, null);
+        if (isEmpty()) {
+            size = 0;
+            first = newNode;
+            last = first;
+        } else {
+            last.next = newNode;
+            newNode.previous = last;
+            last = newNode;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == 0) {
+            addFirst(value);
+        } else {
+            if (index == size) {
+                addToEnd(value);
+            } else {
+                if (isIndexExist(index)) {
+                    Node<T> newNode = new Node<>(null,value,null);
+                    Node<T> oldNode = findForIndex(index);
+                    Node<T> oldPrev = findForIndex(index - 1);
+                    if (oldNode.element != null) {
+                        newNode.previous = oldNode.previous;
+                        newNode.next = oldNode;
+                        oldNode.previous = newNode;
+                        oldPrev.next = newNode;
+                        size++;
+                    }
+                }
+            }
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        if (list == null) {
+            throw new NullPointerException();
+        }
+        for (T one: list) {
+            add(one);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
+        if (isIndexExist(index)) {
+            return findForIndex(index).element;
+        }
         return null;
     }
 
     @Override
     public T set(T value, int index) {
+        if (isIndexExist(index)) {
+            Node<T> requiredNode = findForIndex(index);
+            T oldElement = requiredNode.element;
+            requiredNode.element = value;
+            return oldElement;
+        }
         return null;
     }
 
     @Override
     public T remove(int index) {
+        if (isIndexExist(index)) {
+            Node<T> requiredNode = findForIndex(index);
+            if (index == 0 && size < 2) {
+                size--;
+                first = null;
+                last = null;
+                requiredNode.element = null;
+                return null;
+            }
+            if (index == 0) {
+                first = findForIndex(index + 1);
```

Comment:
you should unlink node, that you removed 
in your case first.prev still pointing at your "removed" node